azext_edge/tests/edge/checks/conftest.py

```diff
@@ -141,12 +141,11 @@ def assert_dict_props(path: str, expected: str, obj: Dict[str, str]):
     val = obj
     for key in path.split("/"):
         val = val[key]
-    if isinstance(val, list):
+
+    if val != expected:
+        if isinstance(val, dict):
+            val = val.values()
```

Comment:
only nit here is that we no longer actually hard assert `expected == val` and we lose structured assertions for `list` types (even though it's the same as `val in dict.values()`), I think the code will have the same result but in my mind it decreases readability.

Was there an error that was causing this check to fail without this change?

Reply:
before this change, we only check if `expect` is one of the elements in `val` when val is list/dict, but didn't check if val actually equals to expected. That would cause the failure when the val is list and == expected.